src/class-validator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Validator class
+ *
+ * @package Parsely
+ * @since   3.9.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+/**
+ * Contains a variety of validation functions.
+ *
+ * @since 3.9.0
+ */
+class Validator {
```

Comment:
Rather than just true / false, I think we should return a `WP_Error` object when the value is invalid. That way, for validations where we have multiple checks, we can provide more specific guidance on what the problem is (e.g. length vs allowed characters), both for end users and also in automated tests.

Not a blocker since we're just extracting existing code but something to consider.

Reply:
I agree. We can enhance this when we improve validation.